### ISSUE TITLE
Any inferred for `not x` where `x: Any`

### ISSUE THREAD

**Issue by jelle-openai** [astral-sh/ty#3572]
### Summary

Unary `not` should always produce `bool`, even when its operand is `Any`.
ty instead infers Any, making it miss downstream type errors.

## Snippet

```python
from typing import Any, reveal_type


def check(value: Any) -> None:
    result = reveal_type(not value)
    result.nonexistent() # false negative
```
